File: Qin/Qin-ios/Qin-ios/Util/Model/Professor.swift
//
//  Professor.swift
//  Qin-ios
//
//  Created by 杨璨 on 2020/2/29.
//  Copyright © 2020 canyang. All rights reserved.
//

import Foundation
import RealmSwift

public class Professor: Object {
    @objc dynamic var ID = ""
    @objc dynamic var Name = ""
    
    public static func GetAll(refresh: Bool){
         if refresh == true {
             var DatabaseResults = [[String:Any?]]()
             var getDBR = [[String:Any?]]()
             repeat {
                 getDBR = [[String:Any?]]()
                 let url = DatabaseHelper.LeancloudAPIBaseURL + "/1.1/classes/Professor?order=ID&limit=1000&skip=\(DatabaseResults.count)"
                 if let response = DatabaseHelper.LCSearch(searchURL: url) {
                     getDBR = response["results"] as! [[String:Any?]]
                     DatabaseResults.append(contentsOf: getDBR)
                 }
             } while getDBR.count != 0
             autoreleasepool {
                 let realm = try! Realm()
                 for checkLog in DatabaseResults {
                     realm.beginWrite()
                     realm.create(Professor.self, value: [
                         "ID" : checkLog["ID"] as! String,
                         "Name" : checkLog["Name"] as! String])
                     try! realm.commitWrite()
                 }
             }
         }
    }
}

File: Qin/Qin-Android/app/src/main/java/edu/bfa/ss/qin/Util/Professor.java
package edu.bfa.ss.qin.Util;

import org.json.JSONArray;
import org.json.JSONObject;

import java.util.ArrayList;

import io.realm.Realm;
import io.realm.RealmObject;

import static edu.bfa.ss.qin.Util.DatabaseHelper.LeancloudAPIBaseURL;

public class Professor extends RealmObject {

    public String ID;
    public String Name;

    public static void GetAll(boolean refresh) {
        Realm realm = Realm.getDefaultInstance();
        realm.beginTransaction();
        if (refresh == true) {
            try {
                ArrayList<Professor> items = new ArrayList<Professor>();
                JSONArray DatabaseResults = new JSONArray();
                do {
                    String url = LeancloudAPIBaseURL + "/1.1/classes/Professor?order=ID&limit=1000&skip=" + items.size();
                    JSONObject response = DatabaseHelper.LCSearch(url);
                    DatabaseResults = response.getJSONArray("results");
                    for (int i = 0; i < DatabaseResults.length(); i++) {
                        JSONObject checkLog = DatabaseResults.getJSONObject(i);
                        Professor newItem = realm.createObject(Professor.class);
                        newItem.ID = checkLog.getString("ID");
                        newItem.Name = checkLog.getString("Name");
                        items.add(newItem);
                    }
                } while (DatabaseResults.length() != 0);
            } catch (Exception ignored) {
                realm.cancelTransaction();
                realm.close();
                return;
            }
        }
        realm.commitTransaction();
        realm.close();
    }
}
